Dockerfile and docker-compose

angelbarrera92 opened the pull request:
Working version solving issue #1 
With those files, you can run the app locally using sqlite *(as you use locally)*.
```
docker build -t trade-motors:0.1.0 .
docker run -d -p 8000:8000 trade-motors:0.1.0
```
Or you can use `docker-compose.yml` file which will run a mysql server along with the web app. 
```
docker-compose up
```

haroon-sheikh commented:
Thanks @angelbarrera92 for the PR. Do you think it's a good idea to use alpine version of docker images instead? i.e. python:2.7-alpine

haroon-sheikh commented on src/gp_cars/settings/docker.py:
```diff
@@ -0,0 +1,156 @@
+        'NAME': 'trademotors',
+        'USER': 'trademotors',
+        'PASSWORD': 'sitture2',
+    }
```
Do these now need to come from environment variables?

angelbarrera92 replied:
I can add `RUN python manage.py test` on Dockerfile. This way the image will not be built if there are test failing.
Yes, you are right. The DB Name, user and password should go on environment variables.

haroon-sheikh commented on docker-compose.yml:
```diff
@@ -0,0 +1,25 @@
+      - MYSQL_DATABASE=trademotors
+      - MYSQL_USER=trademotors
+      - MYSQL_PASSWORD=sitture2
+      - MYSQL_ROOT_HOST=0.0.0.0
```
how about if we store the shared variables in a `.env` file and reference in the file. E.g. https://docs.docker.com/compose/environment-variables/#setting-environment-variables-in-containers

angelbarrera92 replied:
Sounds Good!

angelbarrera92 commented:
The DB parameters have been externalized and I also added a test command on image building process.

angelbarrera92 commented:
Added `.env` file and modified `docker-compose.yml`

About alpine tag... I have some troubles with the mysql client and I think is better to use debian base image during development.

haroon-sheikh commented:
it looks good to merge now. Do you want me to merge this now or wait until tomorrow?

haroon-sheikh commented:
@angelbarrera92 can you add a section in README.md about building/running with Docker?

angelbarrera92 commented:
@haroon-sheikh I will add docker section in README.md. I will do a push ASAP.

angelbarrera92 commented:
Done. Docker info in README.md

angelbarrera92 commented:
@haroon-sheikh it's all ok?
